concept/type/impl/ThingTypeImpl.java

```diff
@@ -379,49 +439,53 @@ FunctionalIterator<AttributeTypeImpl> overriddenOwns(boolean onlyKey, boolean tr
 
     @Override
     public Forwardable<AttributeType, Order.Asc> getOwns() {
-        return getOwns(false);
+        return getOwns(set());
     }
 
     @Override
     public Forwardable<AttributeType, Order.Asc> getOwnsExplicit() {
-        return getOwnsExplicit(false);
+        return getOwnsExplicit(set());
     }
 
     @Override
-    public Forwardable<AttributeType, Order.Asc> getOwns(AttributeType.ValueType valueType) {
-        return getOwns(valueType, false);
+    public Forwardable<AttributeType, Order.Asc> getOwns(Set<TypeQLToken.Annotation> annotations) {
+        return iterateSorted(graphMgr.schema().ownedAttributeTypes(vertex), ASC)
+                .mapSorted(v -> (AttributeType) AttributeTypeImpl.of(graphMgr, v), attr -> ((AttributeTypeImpl) attr).vertex, ASC)
+                .filter(v -> ownsHasAnnotations((AttributeTypeImpl) v, annotations));
     }
 
-    @Override
-    public Forwardable<AttributeType, Order.Asc> getOwnsExplicit(AttributeType.ValueType valueType) {
-        return getOwnsExplicit(valueType, false);
+    boolean ownsHasAnnotations(AttributeTypeImpl attributeType, Set<TypeQLToken.Annotation> annotations) {
+        return getSupertypes().anyMatch(ownerType -> attributeType.getSupertypes().anyMatch(ownedType -> {
+            TypeEdge edge;
+            edge = ownerType.vertex.outs().edge(OWNS, ownedType.vertex);
+            if (edge == null) edge = ownerType.vertex.outs().edge(OWNS_KEY, ownedType.vertex);
+            return edge != null && edge.annotations().containsAll(annotations);
+        }));
```

Comment:
Function to filter on annotations.
Not very efficient because we apply this to every type returned by `getOwns`/`getOwners`